src/nbl/asset/interchange/CGLTFLoader.cpp

```diff
@@ -15,10 +15,20 @@
 #define VERT_SHADER_COLOR_CACHE_KEY "nbl/builtin/shader/loader/gltf/vertex_color.vert"
 #define VERT_SHADER_NO_UV_COLOR_CACHE_KEY "nbl/builtin/shader/loader/gltf/vertex_no_uv_color.vert"
 
+//! created and cached from above shaders
+#define VERT_SHADER_SKINNED_UV_CACHE_KEY "nbl/builtin/shader/loader/gltf/skinned/vertex_uv.vert"
+#define VERT_SHADER_SKINNED_COLOR_CACHE_KEY "nbl/builtin/shader/loader/gltf/skinned/vertex_color.vert"
+#define VERT_SHADER_SKINNED_NO_UV_COLOR_CACHE_KEY "nbl/builtin/shader/loader/gltf/skinned/vertex_no_uv_color.vert"
+
 #define FRAG_SHADER_UV_CACHE_KEY "nbl/builtin/shader/loader/gltf/fragment_uv.frag"
 #define FRAG_SHADER_COLOR_CACHE_KEY "nbl/builtin/shader/loader/gltf/fragment_color.frag"
 #define FRAG_SHADER_NO_UV_COLOR_CACHE_KEY "nbl/builtin/shader/loader/gltf/fragment_no_uv_color.frag"
 
+//! created and cached from above shaders
+#define FRAG_SHADER_SKINNED_UV_CACHE_KEY "nbl/builtin/shader/loader/gltf/skinned/fragment_uv.frag"
+#define FRAG_SHADER_SKINNED_COLOR_CACHE_KEY "nbl/builtin/shader/loader/gltf/skinned/fragment_color.frag"
+#define FRAG_SHADER_SKINNED_NO_UV_COLOR_CACHE_KEY "nbl/builtin/shader/loader/gltf/skinned/fragment_no_uv_color.frag"
```

Comment:
skinned fragment shader doesn't differ from non skinned

Reply:
you have `skinned` in the path for caching, they are created in the constructor

Reply:
but why is the shader code different at all for a frag shader?

Reply:
actually you are right, we don't have to do this with fragment shader

Reply:
I did it for future use but now as I think of it, it's not gonna change anyway

Reply:
corrected in following commit